src/Connection.php

```diff
@@ -0,0 +1,542 @@
+    }
+
+    public function __destruct()
+    {
+        odbc_close($this->connection);
+    }
+
+    public function quoteIdentifier(string $value): string
+    {
+        $q = '"';
+        return ($q . str_replace("$q", "$q$q", $value) . $q);
+    }
+
+    /**
+     * Returns information about table:
+     *  - name
+     *  - bytes
+     *  - rows
+     */
+    public function describeTable(string $schemaName, string $tableName): array
+    {
+        $tables = $this->fetchAll(sprintf(
+            'SHOW TABLES LIKE %s IN SCHEMA %s',
+            "'" . addslashes($tableName) . "'",
+            $this->quoteIdentifier($schemaName)
+        ));
+
+        foreach ($tables as $table) {
+            if ($table['name'] === $tableName) {
+                return $table;
+            }
+        }
+
+        throw new BaseException("Table $tableName not found in schema $schemaName");
+    }
+
+    public function describeTableColumns(string $schemaName, string $tableName): array
+    {
+        return $this->fetchAll(sprintf(
+            'SHOW COLUMNS IN %s.%s',
+            $this->quoteIdentifier($schemaName),
+            $this->quoteIdentifier($tableName)
+        ));
+    }
+
+    public function getTableColumns(string $schemaName, string $tableName): array
+    {
+        return array_map(function ($column) {
+            return $column['column_name'];
+        }, $this->describeTableColumns($schemaName, $tableName));
+    }
+
+    public function getTablePrimaryKey(string $schemaName, string $tableName): array
+    {
+        $cols = $this->fetchAll(sprintf(
+            'DESC TABLE %s.%s',
+            $this->quoteIdentifier($schemaName),
+            $this->quoteIdentifier($tableName)
+        ));
+        $pkCols = [];
+        foreach ($cols as $col) {
+            if ($col['primary key'] !== 'Y') {
+                continue;
+            }
+            $pkCols[] = $col['name'];
+        }
+
+        return $pkCols;
+    }
+
+    public function query(string $sql, array $bind = []): void
+    {
+        try {
+            $stmt = odbc_prepare($this->connection, $sql);
+            odbc_execute($stmt, $this->repairBinding($bind));
+            odbc_free_result($stmt);
+        } catch (\Throwable $e) {
+            (new ExceptionHandler())->handleException($e, $sql);
+        }
+    }
+
+    public function fetchAll(string $sql, array $bind = []): array
+    {
+        try {
+            $stmt = odbc_prepare($this->connection, $sql);
+            odbc_execute($stmt, $this->repairBinding($bind));
+            $rows = [];
+            while ($row = odbc_fetch_array($stmt)) {
+                $rows[] = $row;
+            }
+            odbc_free_result($stmt);
+            return $rows;
+        } catch (Throwable $e) {
+            throw new RuntimeException(sprintf('Error "%s" while executing query "%s"', $e->getMessage(), $sql));
+        }
+    }
+
+    public function fetch(string $sql, array $bind, callable $callback): void
```

Comment:
Tady předtím bylo zpracování vyjímky. Dal si to pryč naschvál ?
https://github.com/keboola/php-db-import/blob/f3c099842ea8fbfe824a31d782be26c0bf7fe8e4/src/Snowflake/Connection.php#L194

Reply:
Rozhodně ne, díky!

Reply:
Ne, to nebylo naschvál. Díky. 

Nechápu, jak se to mohlo stát.

Reply:
Už chápu - v tom commitu, kde to byl jen copypaste (ten první) je totiž nějaká jiná verze než je teď master :( Asi jsem neměl fetchnulo 🤦‍♀️

Reply:
Zpracování je tam zpátky.